### PR TITLE
Support collections in index functions

### PR DESCRIPTION
**Pull request by matthchr**

### CODE REVIEW

**Review by theunrepentantgeek — APPROVED** [review]
Nice and simple.

File: v2/tools/generator/internal/functions/index_registration_function.go [hunk 1]
```diff
@@ -88,30 +91,19 @@ func (f *IndexRegistrationFunction) AsFunc(
...
-	nilGuards := make([]dst.Stmt, 0, len(f.propertyChain))
-	for _, prop := range f.propertyChain {
-		propNames = append(propNames, prop.PropertyName().String())
-		intermediateSelector := astbuilder.Selector(specSelector, propNames...)
-		nilGuards = append(nilGuards, astbuilder.ReturnIfNil(intermediateSelector, astbuilder.Nil()))
+	var stmts []dst.Stmt
+	if f.isIndexSingleValue() {
+		stmts = f.simple(specSelector)
+	} else {
+		stmts = f.complex(specSelector)
```
[the remark this answers is not in the document]
Comment (matthchr) [reply v2/tools/generator/internal/functions/index_registration_function.go:98]: Good suggestion, done

**Pull request merged**